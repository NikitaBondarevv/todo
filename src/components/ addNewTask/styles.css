.addTask {
  background: #fff;
  width: 244px;
  font-size: 22px;
  line-height: 55px;
  text-align: center;
  text-decoration: underline;
  box-shadow: 0px 1px 7px #000;
  margin: 0 0 14px 59px;
}

.editableText {
  height: 45px;
  width: 234px;
  font-size: 22px;
}
