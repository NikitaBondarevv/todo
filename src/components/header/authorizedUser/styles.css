.user {
  position: relative;
}

.userName {
  display: block;
  margin: 0 0 0 255px;
  font-size: 26px;
  line-height: 24px;
  color: #fff;
  background: url(./images/user.png) no-repeat 60px 0;
  min-width: 90px;
  text-decoration: none;
}

.userName:hover {
  color: #060e9a;
  background: url(./images/hoverUser.png) no-repeat 60px 0;
  text-decoration: underline;
}

.menu {
  background: rgba(255,255,255, 0.3);
  flex-direction: column;
  padding: 5px 17px 0;
  position: absolute;
  right: -5px;
  top: 26px;
  display: none;
}

.user:hover .menu {
  display: flex;
}

.menu a {
  text-decoration: none;
  color: #fff;
  font-size: 26px;
  line-height: 24px;
  white-space: nowrap;
  margin: 0 0 5px;
}

.menu a:hover {
  color: rgb(169, 169, 169);
}
