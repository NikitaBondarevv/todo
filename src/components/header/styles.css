header {
  font-size: 26px;
  line-height: 26px;
  display: grid;
  grid-template-columns: 0.1fr 1fr auto;
  align-items: center;
  padding: 23px 0 0;
  margin: 0 0 60px;
}

header .list {
  list-style: none;
  margin: 0;
  padding: 0;
  display: flex;
  column-gap: 68px;
}

.list a {
  color: #fff;
  text-decoration: none;
  position: relative;
}

.list a:hover {
  color: #060e9a;
  text-decoration: underline;
}

.list a::before {
  background: url(./images/menu.png) no-repeat;
  content: "";
  display: block;
  width: 25px;
  height: 25px;
  right: -30px;
  bottom: 2px;
  position: absolute;
}

.list a:hover::before {
  background: url(./images/hoverMenu.png) no-repeat;
  content: "";
  display: block;
  width: 25px;
  height: 25px;
  right: -30px;
  bottom: 2px;
  position: absolute;
}

.list .home:hover::before {
  background-position: 2px 2px;
}

.list .taskList:hover::before {
  background-position: -23px 2px;
}

.list .contacts:hover::before {
  background-position: -46px 2px;
}

.list .taskList::before {
  background-position: -25px 0;
}

.list .contacts::before {
  background-position: -52px 0;
}

.createUser {
  margin: 0 0 0 220px;
  font-size: 26px;
  line-height: 24px;
  color: #fff;
  text-decoration: none;
  background: url(./images/user.png) no-repeat;
  background-position: 140px 0;
  width: 165px;
}

.createUser:hover {
  color: #060e9a;
  background: url(./images/hoverUser.png) no-repeat;
  background-position: 140px 0;
  text-decoration: underline;
}
