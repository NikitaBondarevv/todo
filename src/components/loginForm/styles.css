.loginForm {
  display: flex;
  flex-direction: column;
  width: 675px;
  height: 406px;
  margin: 0 auto 100px;
  background: rgb(7 21 147 / 50%);
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 60px 80px;
}

.loginForm input,
.loginForm button[type=submit] {
  font: 24px/24px "MyriadProLight";
  padding: 5px 5px 5px 30px;
  width: 247px;
  height: 55px;
  border-radius: 7px;
  margin: 0 0 15px 0;
  border: 1px solid #fff;
  font-style: italic;
  color: #000;
}

.loginForm input::placeholder {
  color: #000;
}

.loginForm button[type=submit] {
  padding: 0;
  margin: 0;
  background: #0ff;
  color: #fff;
  height: 53px;
  width: 143px;
  border: 1px solid #b1bad8;
  box-shadow: 0 0 5px #b1bad8;
}

.loginForm button[type=submit]:disabled {
  background: #316262;
}

.password {
  position: relative;
}

.loginForm .password::before {
  background: url(./images/form-elements.png) no-repeat;
  content: "";
  width: 17px;
  position: absolute;
  z-index: 2;
  left: 250px;
  top: 23px;
}

.loginForm .password::before {
  background-position: 0 -33px;
  height: 18px;
}

.error {
  color: red;
  font-size: 25px;
}
