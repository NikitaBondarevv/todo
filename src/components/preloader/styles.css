@keyframes loader {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

.preloader {
  margin: 0 auto;
  border: 20px solid #086dc0;
  border-top-color: transparent;
  border-radius: 50%;
  animation: loader 1s linear infinite;
}
