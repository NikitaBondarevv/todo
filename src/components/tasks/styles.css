@font-face {
  font-family: "MyriadProBold";
  src: url(../../../fonts/MyriadProBold.ttf);
}

.tasksList {
  margin: 34px 0 52px;
  padding: 0 15px 0;
  font-size: 24px;
  line-height: 39px;
  color: #fff;
  list-style: none;
  counter-reset: my-awesome-counter;
}

.tasksList li {
  display: flex;
  margin: 0 0 15px;
  height: 35px;
  min-width: 35px;
  counter-increment: my-awesome-counter;
  padding: 0 0 0 65px;
  white-space: nowrap;
}

.tasksList .updateTask:hover {
  text-decoration: underline;
  cursor: pointer;
}

.tasksList li::before {
  content: counter(my-awesome-counter) ". ";
  font-weight: bold;
}

.tasksList p {
  margin: 0;
  padding: 0 0 0 50px;
}

.tasksList .waiting {
  background: url(./images/waiting.png) no-repeat;
}

.tasksList .done {
  text-decoration: line-through;
  background: url(./images/done.png) no-repeat;
}

.tasksList .progress {
  background: url(./images/in-progress.png) no-repeat;
  background-position: 0 5px;
}

.tasksList button {
  margin: 0 7px;
  background: url(./images/buttons.png) no-repeat;
  width: 38px;
  height: 38px;
  cursor: pointer;
  border: none;
}

.tasksList button:hover {
  background: url(./images/hoveringButtons.png) no-repeat;
}

.tasksList .inProgress {
  background-position: -53px 0;
}

.tasksList .delete {
  background-position: -105px 0;
}

.tasksList .inProgress:hover {
  background-position: -53px 0;
}

.tasksList .delete:hover {
  background-position: -105px 0;
}

.buttons {
  margin: 0 0 0 30px;
}

.active a {
  box-shadow: 0 -2px 2px rgb(0 0 0 / 50%);
  font-weight: bold;
  background: #303994
}

.activeContent {
  background: #303994
}
