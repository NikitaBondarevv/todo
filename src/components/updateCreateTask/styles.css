.updateCreateTask {
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 39px 29px 70px;
  background: rgba(0, 0, 255, .3);
  width: 777px;
  color: #fff;
  margin: 0 auto 100px;
  display: flex;
  flex-direction: column;
  row-gap: 19px;
  padding: 59px;
  position: absolute;
  top: 400px;
  left: 0;
  width: 717px;
}

.updateCreateTask .title,
.updateCreateTask .description {
  width: 257px;
  outline: none;
  font: 24px/40px "MyriadProRegular";
  padding: 0 0 0 16px;
}

.updateCreateTask .description {
  height: 70px;
  resize: none;
  margin: 0 0 96px;
}

.updateCreateTask input[type=button] {
  padding: 0;
  margin: 0;
  background: #0ff;
  color: #fff;
  height: 53px;
  width: 143px;
  border: 1px solid #b1bad8;
  box-shadow: 0 0 5px #b1bad8;
  border-radius: 6px;
  font: 24px/24px "MyriadProLight" ;
  font-style: italic;
  margin: 0 0 120px;
}
