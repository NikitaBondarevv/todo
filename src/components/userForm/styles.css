.loginForm {
  padding: 20px;
  min-height: 250px;
  background: rgb(7 21 147 / 50%);
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  border: 1px solid #fff;
  width: 795px;
  margin: 0 auto;
}

.loginForm ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

.loginForm li {
  position: relative;
}

.loginForm input {
  font: 24px/24px "MyriadProLight";
  padding: 5px 35px 5px 30px;
  width: 212px;
  height: 55px;
  border-radius: 7px;
  margin: 0 20px 15px 0;
  border: 1px solid #fff;
  font-style: italic;
}

.loginForm .email::before,
.loginForm .name::before,
.loginForm .password::before {
  background: url(./images/form-elements.png) no-repeat;
  content: "";
  width: 17px;
  position: absolute;
  z-index: 2;
  left: 250px;
  top: 29px;
}

.loginForm .email::before {
  height: 13px;
}

.loginForm .name::before {
  background-position: 0 -15px;
  height: 17px;
}

.loginForm .password::before {
  background-position: 0 -33px;
  height: 18px;
}

.loginForm  .error {
  border: 3px solid #f17777;
}

.loginForm .success {
  border: 3px solid green;
}

.loginForm input::placeholder {
  color: #000;
}

.loginForm input[type=submit] {
  padding: 0;
  margin: 0;
  background: #0ff;
  color: #fff;
  height: 53px;
  width: 143px;
  border: 1px solid #b1bad8;
  box-shadow: 0 0 5px #b1bad8;
  cursor: pointer;
}

.loginForm input:disabled {
  background-color: rgb(255 255 255 / 50%);
  cursor: default;
}

.info {
  background: #f9f8c9;
  border-radius: 4px;
  height: 55px;
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  bottom: 23px;
  left: 310px;
  padding: 0 25px;
}

.info::before {
  background: url(./images/msg-figure.png) no-repeat;
  content: "";
  width: 14px;
  position: absolute;
  height: 12px;
  left: -12px;
  top: 15px;
}

.textInfo {
  color: #ed2323;
  font-size: 22px;
  line-height: 24px;
  margin: 0 0 0 8px;
}
