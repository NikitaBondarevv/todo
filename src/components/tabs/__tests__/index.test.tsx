import { render } from '@testing-library/react'

import { Tabs } from '..'
import styles from './styles.css'

describe('<Tabs />', () => {
  test('should match snapshot', () => {
    const { asFragment } = render(
    <Tabs loading={false}>
      <div>test</div>
    </Tabs>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render <Preloader /> if loading = true', () => {
    const { container } = render(<Tabs children={[]} loading />)

    const preloader = container.querySelector(`.${styles.preloader}`)

    expect(preloader).toBeInTheDocument()
  })
})
