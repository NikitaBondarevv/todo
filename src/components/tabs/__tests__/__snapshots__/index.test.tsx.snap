// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    <nav>
      <ul
        class="list"
      >
        <li
          class="active"
        >
          <a>
            <br />
            <time
              class="dates"
            >
              27 March
            </time>
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="activeContent"
      id="0"
    >
      test
    </div>
  </div>
</DocumentFragment>
`;
